doc/old_and_new_proposal.md

```diff
@@ -108,18 +178,453 @@ addition, every volume gets a "weight" (so far, based on the
+    defined is disabled, the `min_size` of that another volume will be increased
+    by the `min_size` of this disabled volume.
+  * `fallback_for_desired_size`. Same than before, but for `desired_size`.
+  * `fallback_for_max_size`. Same than before, but for `max_size`.
+  * `fallback_for_max_size_lvm`. Same than before, but for `max_size_lvm`.
+  * `fallback_for_weight`. Same than before, but for the volume weight.
+
+Some options will only apply if the chosen filesystem type for the volume is
+Btrfs, in some cases with the same name and meaning that in the old
+`control.xml` format. The main difference is that now the setting will apply to
+the volume in which it's included, not necessarily to the root ("/") one. In the
+expert partitioner, if a Btrfs filesystem is created and assigned to the mount
+point of the volume, these settings will also be used to suggest the filesystem
+options.
+
+  * `snapshots`. Whether snapshots should be activated. Possible values are
+    "true", "false" and "optional", with "false" being the default if not
+    present and "optional" meaning the user can decide via UI.
+  * `default_snapshots`. If the previous value is "optional", whether to
+    enable the option by default.
+  * `snapshots_size`. Similar to `btrfs_increase_percentage` in the
+    yast-storage format, but slightly more flexible. If it's a size, the initial
+    and maximum sizes for the volume will be increased according if snapshots
+    are being used. If it's a number, it will be used as a percentage of the
+    original sizes (just like the original `btrfs_increase_percentage`).
+  * `subvolumes`. Equivalent to the previous option that used to apply only to
+    "/".
+  * `btrfs_default_subvolume`. Same than before. NOTE: keeping this as a
+    separate setting instead of making it just another subvolume in the list
+    above is debatable.
+
+And finally there is an option that deserves a sligthly more detailed
+explanation.
+
+  * `disable_order`. Volumes with some value here will be disabled (or snapshots
+    deactivated) if needed to make the initial proposal. See detailed
+    explanation below.
+
+Before any user interaction, an initial proposal with the default settings is
+calculated. If YaST is not able to make space for all the volumes required by
+those default settings, it will perform new attempts altering the settings. For
+that, it will follow the `disable_order` for each volume with that field.
+
+In the first iteration, it will look for the lowest number there. If
+`adjust_by_swap` is optional in that volume and enabled, it will disable it. If
+that is not enough and snapshots are optional but enabled, it will disable them
+and try again (assuming Btrfs is being used). If that's still not enough, it
+will disable the whole volume if it's optional.
+
+If that's not enough, it will keep those settings and look for the next volume
+with some value in `disable_order` to perform the same operation in a cumulative
+fashion.
+
```

Comment:
The number and order of attempts will depend on the control file. But in the SLES example provided in this document it would be still reasonable:

* First disabling /home
* Second also not trying to enlarge swap (if it makes sense)
* Third without snapshots.